samza-test/src/main/java/org/apache/samza/test/framework/StreamAssert.java

```diff
@@ -20,195 +20,33 @@
-
-    private final String id;
-    private final boolean checkEachTask;
-    private final Collection<M> expected;
-
-    private transient Timer timer = new Timer();
-    private transient List<M> actual = Collections.synchronizedList(new ArrayList<>());
-    private transient TimerTask timerTask = new TimerTask() {
-      @Override
-      public void run() {
-        check();
-      }
-    };
-
-    CheckAgainstExpected(String id, Collection<M> expected, boolean checkEachTask) {
-      this.id = id;
-      this.expected = expected;
-      this.checkEachTask = checkEachTask;
-    }
-
-    @Override
-    public void init(Config config, TaskContext context) {
-      final SystemStreamPartition ssp = Iterables.getFirst(context.getSystemStreamPartitions(), null);
-      if (ssp == null ? false : ssp.getPartition().getPartitionId() == 0) {
-        final int count = checkEachTask ? context.getSamzaContainerContext().taskNames.size() : 1;
-        LATCHES.put(id, new CountDownLatch(count));
-        timer.schedule(timerTask, TIMEOUT);
-      }
-    }
-
-    @Override
-    public void apply(M message, MessageCollector messageCollector, TaskCoordinator taskCoordinator) {
-      actual.add(message);
-
-      if (actual.size() >= expected.size()) {
-        timerTask.cancel();
-        check();
-      }
-    }
-
-    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
-      in.defaultReadObject();
-      timer = new Timer();
-      actual = Collections.synchronizedList(new ArrayList<>());
-      timerTask = new TimerTask() {
-        @Override
-        public void run() {
-          check();
-        }
-      };
-    }
-
-    private void check() {
-      final CountDownLatch latch = LATCHES.get(id);
-      try {
-        assertThat(actual, Matchers.containsInAnyOrder((M[]) expected.toArray()));
-      } finally {
-        latch.countDown();
-      }
-    }
-  }
-
+public class StreamAssert {
   /**
    * Util to assert  presence of messages in a stream with single partition in any order
    *
+   * @param collectionStream represents the actual stream which will be consumed to compare against expected list
    * @param expected represents the expected stream of messages
    * @param timeout maximum time to wait for consuming the stream
+   * @param <M> represents the type of Message in the stream
    * @throws InterruptedException when {@code consumeStream} is interrupted by another thread during polling messages
-   *
    */
-  public void containsInAnyOrder(final List<M> expected, Duration timeout) throws InterruptedException {
-    Preconditions.checkNotNull(collectionStream,
-        "This util is intended to use only on CollectionStream)");
+  public static <M> void containsInAnyOrder(CollectionStream<M> collectionStream, final List<M> expected, Duration timeout)
+      throws InterruptedException {
+    Preconditions.checkNotNull(collectionStream, "This util is intended to use only on CollectionStream");
```

Comment:
Should this precondition message be updated? If you are going to validate the "collectionStream", should you check against the other arguments as well?